content/app/commands/highlight-text-command.js

```diff
@@ -0,0 +1,111 @@
+define("content/app/commands/highlight-text-command", function(require, exports, module) {
+
+	// imports
+	var Event =require("content/app/events/event");
+	var AbstractCommand = require("content/app/commands/abstract-command");
+	var CommitPageHelper = require("content/app/github/commit-page-helper");
+	var WindowHelper = require("content/app/github/window-helper");
+	var Box = require("content/app/models/box");
+
+	/**
+	 * This command is used for selecting text in files.
+	 */
+	var HighlightTextCommand = Class(AbstractCommand, {
+
+		/**
+		 * HighlightTextCommand constructor.
+		 */
+		constructor : function () {
+			HighlightTextCommand.$super.call(this);
+
+			this._onSelectHandler = this._onSelect.bind(this);
+			this._highlights = [];
+		},
+
+		/**
+		 * Override. Run command.
+		 */
+		run : function () {
+			WindowHelper.on(Event.SELECT, this._onSelectHandler);
+		},
+
+		/**
+		 * Override. Cancel command.
+		 */
+		cancel : function () {
+			this._clearHightlights();
+			WindowHelper.off(Event.SELECT, this._onSelectHandler);
+		},
+
+		/**
+		 * Select text event handler.
+		 * @param  {Event} evt
+		 */
+		_onSelect : function (evt) {
+			this._clearHightlights();
+			this._highlightText(evt.data);
+		},
+
+		/**
+		 * Create highlights for provided text.
+		 * @param  {String} text
+		 */
+		_highlightText : function (text) {
+			if (text && text.length) {
+				var ranges = CommitPageHelper.findTextRangesInFiles(text);
```

Comment:
I do not think that we need to select words in all files, just in that one where user has double clicked. I do not see scenario where you select variable in one file and you are interested in variable from another file with same name.